### PR TITLE
Fix/172 css grid restyling

### LINKED ISSUE #172: CSS Grid Restyling
In SPAR's front end there are several places where `!important` is used with the intention of having a default style.
However, using `!important` is considered as a [bad practice](https://element.how/why-you-should-avoid-using-important-in-css/#:~:text=important%20rule%20in%20CSS%20is,unexpected%20behavior%20in%20your%20website.). 

In `frontend/src/theme/components-overrides.scss` we specified `--grid` to have 0 padding, this style will override all the other padding style a developer wishes to add, which is frustrating to work with. It is evident that this practice should be corrected ASAP so it does not affect future development. 

This issue is to tackle the issues appeared with removal of the `!important` styling for Carbon's `Grid` component.

**Before** Padding sticks out:
Image

**After** Exactly 40px padding as designed: 
Image

### PR DESCRIPTION
**Pull request by craigyu**
# Description
This PR removes the usage of `!important` for `Grid` and `FlexGrid` component. The front end was built to accommodate this `!important` override in the beginning so a lot of components need to be refactored in another issue. 
To keep each PR small(fairly small), this one only refactored the Collection Step of the registration flow as an example.
To see the reason for this change please see #172 for detail. 

### Summary of changes
- remove `important` for grid css in the component override file
- refactor css styling for `CollectionStep` and `SeedlotRegistrationForm`
- add button text and icon customization feature to `SubmitModal` component

### Note:

Fixes #172 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Local testing

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-174-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-174-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-174-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx [hunk 1]
```diff
@@ -170,342 +170,340 @@ const CollectionStep = (
...
+                e.target.value
+              );
+            }}
+            hideSteppers
+            disableWheel
+          />
+        </Column>
+      </Row>
+      <Row className="collection-step-row">
+        <Column lg={16}>
```
Comment (mgaseta) [inline frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx:244]: Add other screens sizes to this column.

File: frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx [hunk 2]
```diff
@@ -170,342 +170,340 @@ const CollectionStep = (
...
+                e.target.value
+              );
+            }}
+            hideSteppers
+            disableWheel
+          />
+        </Column>
+      </Row>
+      <Row className="collection-step-row">
+        <Column sm={4} md={8} lg={16} xlg={16}>
```
Comment (mgaseta) [inline frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx:369]: Should we refactor these checkboxes? I don't see the need of not using a map here.

File: frontend/src/components/DescriptionBox/styles.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
+@use '../../theme/variables.scss' as vars;
+
+.description-box-container{
```
[the remark this answers is not in the document]
Comment (craigyu) [reply frontend/src/components/DescriptionBox/styles.scss:3]: added in the same commit as [above](https://github.com/bcgov/nr-spar/pull/174/commits/a5d82d979af9c3c492be3dbd41572c7455039728)

File: frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx (hunk 1 above)
Comment (craigyu) [reply frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx]: [added](https://github.com/bcgov/nr-spar/pull/174/commits/7b4acd38fa5870f845df2610214c176266ad7029)

File: frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx (hunk 2 above)
Comment (craigyu) [reply frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx:369]: We'll have to refactor the entirety of this checkbox field to integrate with the API, the skeleton Jira issue is [here](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-682) let's leave it for now

File: frontend/src/components/DescriptionBox/styles.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  .description-box-header{
+    color: var(--#{vars.$bcgov-prefix}-text-primary);
+    font-size: 1.25em;
```
[the remark this answers is not in the document]
Comment (craigyu) [reply frontend/src/components/DescriptionBox/styles.scss]: [fixed](https://github.com/bcgov/nr-spar/pull/174/commits/a5d82d979af9c3c492be3dbd41572c7455039728)

File: frontend/src/components/DescriptionBox/styles.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+  .description-box-header{
+    color: var(--#{vars.$bcgov-prefix}-text-primary);
+    font-size: 1.25rem;
+  }
+
+  .description-box-description{
+    color: var(--#{vars.$bcgov-prefix}-text-secondary);
+    font-size: 0.875em;
+    margin-top: 0.5em;
```
Comment (mgaseta) [inline frontend/src/components/DescriptionBox/styles.scss:13]: Change to rem here also.

File: frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx (hunk 2 above)
Comment (mgaseta) [reply frontend/src/components/SeedlotRegistrationSteps/CollectionStep/index.tsx:369]: Ok! :)

**Comment by craigyu** [comment]
As @mgaseta pointed out, this PR contains some changes for the parent tree step component which do not fit the scope of this issue. The reason being this task was branched out of the development of parent tree, in the future this practice will be avoided. For "efficiency" sake, these changes will be pushed to `main` this time.

**Review by mgaseta — CHANGES_REQUESTED** [review]
Just another typo to fix and it's good to go!

File: frontend/src/components/DescriptionBox/styles.scss (hunk 5 above)
Comment (craigyu) [reply frontend/src/components/DescriptionBox/styles.scss]: [done](https://github.com/bcgov/nr-spar/pull/174/commits/db858dfdc598dd0bf0a05eafb21755b53c3455b6)

**Review by mgaseta — APPROVED** [review]
LGTM!

**Pull request merged**